lib/Runtime/Library/IntlEngineInterfaceExtensionObject.cpp

```diff
@@ -2590,7 +2610,15 @@ namespace Js
         {
             int curLen = -1;
             const UChar *cur = uenum_unext(available, &curLen, &status);
-            ICU_ASSERT(status, curLen > 0);
+            ICU_ASSERT(status, true);
+            if (curLen == 0)
+            {
+                // OS#17175014: in rare cases, ICU will return U_ZERO_ERROR and a valid `cur` string but a length of 0
```

Comment:
I suppose if you wanted you could also mention/link to the upstream ICU bug as well.
http://bugs.icu-project.org/trac/ticket/13739

Reply:
Done.